x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/TimeoutChecker.java

```diff
@@ -38,7 +40,7 @@
 public class TimeoutChecker implements Closeable {
```

Comment:
@droberts195 Does this change look good to you?

This change to ML is required in order to make the watchdog `Matcher` based instead of `Thread` based.